include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -2686,6 +2766,181 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::
   return cell_transfer.run(cell_transfer_test);
 }
 
+
+
+namespace RepartitioningPolicyTools
+{
+  namespace
+  {
+    template <typename MeshType>
+    unsigned int
+    compute_n_coarse_cells(const MeshType &mesh)
+    {
+      types::coarse_cell_id n_coarse_cells = 0;
+
+      for (const auto &cell : mesh.get_triangulation().active_cell_iterators())
+        if (!cell->is_artificial())
+          n_coarse_cells =
+            std::max(n_coarse_cells, cell->id().get_coarse_cell_id());
+
+      return Utilities::MPI::max(n_coarse_cells, mesh.get_communicator()) + 1;
+    }
+
+
+
+    template <typename MeshType>
+    unsigned int
+    compute_n_global_levels(const MeshType &mesh)
+    {
+      return mesh.get_triangulation().n_global_levels();
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    add_indices(const TriaIterator<CellAccessor<dim, spacedim>> &cell,
+                const internal::CellIDTranslator<dim> &cell_id_translator,
+                IndexSet &                             is_fine)
+    {
+      is_fine.add_index(cell_id_translator.translate(cell));
+
+      if (cell->level() > 0 &&
+          (cell->index() % Utilities::pow<unsigned int>(2, dim)) == 0)
+        add_indices(cell->parent(), cell_id_translator, is_fine);
+    }
+  } // namespace
+
+  template <int dim, int spacedim>
+  LinearAlgebra::distributed::Vector<double>
+  DefaultPolicy<dim, spacedim>::partition(
+    const Triangulation<dim, spacedim> &tria_coarse_in) const
+  {
+    (void)tria_coarse_in; // nothing to do
+
+    return {};
+  }
+
+
+
+  template <int dim, int spacedim>
+  FirstChildPolicy<dim, spacedim>::FirstChildPolicy(
+    const Triangulation<dim, spacedim> &tria_fine)
+    : n_coarse_cells(compute_n_coarse_cells(tria_fine))
+    , n_global_levels(compute_n_global_levels(tria_fine))
+  {
+    const internal::CellIDTranslator<dim> cell_id_translator(n_coarse_cells,
+                                                             n_global_levels);
+    is_fine.set_size(cell_id_translator.size());
+
+    for (const auto &cell : tria_fine.active_cell_iterators())
+      if (cell->is_locally_owned())
+        add_indices(cell, cell_id_translator, is_fine);
+  }
+
+
+
+  template <int dim, int spacedim>
+  LinearAlgebra::distributed::Vector<double>
+  FirstChildPolicy<dim, spacedim>::partition(
+    const Triangulation<dim, spacedim> &tria_coarse_in) const
+  {
+    const auto communicator = tria_coarse_in.get_communicator();
+
+    const internal::CellIDTranslator<dim> cell_id_translator(n_coarse_cells,
+                                                             n_global_levels);
+
+    IndexSet is_coarse(cell_id_translator.size());
+
+    for (const auto &cell : tria_coarse_in.active_cell_iterators())
+      if (cell->is_locally_owned())
+        is_coarse.add_index(cell_id_translator.translate(cell));
+
+    std::vector<unsigned int> owning_ranks_of_ghosts(is_coarse.n_elements());
```

Comment:
Wouldn't a more appropriate name be
```suggestion
    std::vector<unsigned int> owning_ranks_of_coarse_cells(is_coarse.n_elements());
```